### PR TITLE
feat: updated character length to 3 for ENS controller

### PR DESCRIPTION
**Pull request by NidhiKJha**
## Explanation

This PR is to update ens character length from 7 in regex to 3

Fixes: [issue](https://consensyssoftware.atlassian.net/browse/CEUX-1027)

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've communicated my changes to consumers by [updating changelogs for packages I've changed](https://github.com/MetaMask/core/tree/main/docs/processes/updating-changelogs.md)
- [x] I've introduced [breaking changes](https://github.com/MetaMask/core/tree/main/docs/processes/breaking-changes.md) in this PR and have prepared draft pull requests for clients and consumer packages to resolve them

---

> [!NOTE]
> **Medium Risk**
> Changes input validation for ENS names, which can affect what names are accepted/stored across controllers; logic change is small but broadens allowed inputs and could surface edge-case name handling differences.
> 
> **Overview**
> ENS name validation is loosened by updating `normalizeEnsName` to accept `.eth`/`.test` names where the final label is **3+ characters** (previously **7+**), enabling shorter ENS registrations to be stored and resolved.
> 
> Tests are updated to reflect the new minimum length, including coverage that `EnsController.set` now accepts `foo.eth` and that multi-level names like `foo.barbaz.eth` are no longer rejected purely for length; the `controller-utils` changelog is updated accordingly.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 07d92fdb16b833e42143cb02896da17514aee853. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/controller-utils/src/util.ts [hunk 1]
```diff
@@ -541,8 +541,7 @@ export function normalizeEnsName(ensName: string): string | null {
     try {
       const normalized = ensNamehash.normalize(ensName.trim());
       // this regex is only sufficient with the above call to ensNamehash.normalize
-      // TODO: change 7 in regex to 3 when shorter ENS domains are live
-      if (normalized.match(/^(([\w\d-]+)\.)*[\w\d-]{7,}\.(eth|test)$/u)) {
+      if (normalized.match(/^(([\w\d-]+)\.)*[\w\d-]{3,}\.(eth|test)$/u)) {
```
Comment (FrederikBolding) [inline packages/controller-utils/src/util.ts:544]: Can you add a CHANGELOG entry for this change?

Comment (NidhiKJha) [reply packages/controller-utils/src/util.ts:544]: Updated here https://github.com/MetaMask/core/pull/8510/changes/870e66ff74595a4bf787677917977aeb689a64e6

**Review by FrederikBolding — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]
